### PR TITLE
fixed runtime error on loss plotting in section 3.5 and 4 in w2d1

### PR DESCRIPTION
**Pull request by YichengShen**
Fix runtime error.

In section 3.5 and the first code cell in section 4. The loss collected in the training loop are tensors requiring gradients. When plotting the loss, `plt.plot(range(num_epochs), losses)` throws an error when trying to convert this list of tensors requiring gradients to a numpy array.

To fix this, each loss `l` is detached before being appended to the loss list in the training loop.

### CODE REVIEW

**Comment by j6k4m8** [comment]
Super awesome, thank you!

**Review by j6k4m8 — APPROVED** [review]
